engines/http.c

```diff
@@ -377,19 +453,46 @@ static void _add_aws_auth_header(CURL *curl, struct curl_slist *slist, struct ht
 	}
 
 	/* Create the canonical request first */
-	snprintf(creq, sizeof(creq),
-	"%s\n"
-	"%s\n"
-	"\n"
-	"host:%s\n"
-	"x-amz-content-sha256:%s\n"
-	"x-amz-date:%s\n"
-	"x-amz-storage-class:%s\n"
-	"\n"
-	"host;x-amz-content-sha256;x-amz-date;x-amz-storage-class\n"
-	"%s"
-	, method
-	, uri_encoded, o->host, dsha, date_iso, o->s3_storage_class, dsha);
+	if (sse_key[0] != '\0') {
+		sse_key_base64 = _conv_base64_encode(sse_key, sizeof(sse_key) - 1);
+		sse_key_md5_base64 = _gen_base64_md5(sse_key, sizeof(sse_key) - 1);
+		snprintf(creq, sizeof(creq),
+		"%s\n"
```

Comment:
Please indent this and later lines so that it is obvious these are parameters of snprintf

Reply:
done.